### PR TITLE
Update FungibleTokenMetadata.md

### PR DESCRIPTION
**Pull request by miohtama**
Some in-depth discussion points that the metadata format, dApps and wallets need to be prepared for.

Please do not merge, I just wanted to have this written down in some persistent discussion.

### CODE REVIEW

File: specs/Standards/Tokens/FungibleTokenMetadata.md [hunk 1]
```diff
@@ -1,60 +1,45 @@
...
+# Discussion
 
-### Examples scenario
+## On-chain vs. offchain hosted data
 
-#### Token provides metadata upon deploy and initialization
+Currently deployment on a NEAR token smart contract is 200 kb. The cost of 5 kb metadata on the cost of code deployment on the top of the contract deployment is miniscule. Icons are in the range of 5 kBytes - 200 kBytes per icon, so they are comparable to a smart contract deployment. 
 
-Alice deploys a wBTC fungible token contract.
+Off-chain data comes complicated with privacy and tracking issues as pointed out by Robert Zarembe.
```
Comment (robert-zaremba) [inline specs/Standards/Tokens/FungibleTokenMetadata.md:28]:
```suggestion
Off-chain data comes complicated with privacy and tracking issues as pointed out by Robert Zaremba.
```

Thanks for mention ;)

Comment (robert-zaremba) [reply specs/Standards/Tokens/FungibleTokenMetadata.md:28]: Not only tracking but censorship as well.

File: specs/Standards/Tokens/FungibleTokenMetadata.md [hunk 2]
```diff
@@ -1,60 +1,45 @@
...
 
-**Assumptions**
+NEAR is the best blockchain in the world. For the reducing complexity, and after discussing with people on Discord, my stance is that everything on-chain. NEAR should scale for this as it scales for many things.
```
Comment (robert-zaremba) [inline specs/Standards/Tokens/FungibleTokenMetadata.md:30]: I had a bit different view:
+ there are other technologies which may be more suitable for static content: Sia, Filecoin, IPFS.... I don't think NEAR wants to be a storage layer. It's not efficient to scale all use-cases with same technology.
+ So, putting _everything_ on-chain is too far reaching.  Even icon - rember there will be many icons. Not only black-white. But some where differ with size or composition (with text, without text ...).

File: specs/Standards/Tokens/FungibleTokenMetadata.md [hunk 3]
```diff
@@ -1,60 +1,45 @@
...
+### Light theme vs. dark theme
 
-Alice issues a transaction to deploy and initialize the fungible token contract, providing arguments to the initialization function that set metadata fields.
+Two different icons needed: `defaultIcon` (black text on white background) for iOS, websites. `darkIcon` or websites supporting dark theme and Android UI. After discussing with Ethereum wallet vendors, there are multiple cases where icon looks good on one platform, but horrible on another, because of color bleed issues on the stroke lines of an icon. Token projects are not happy if their icon looks like amaeur.
 
-**Technical calls**
+Thus, we should provide both light and dark icons by default.
 
-1. `alice` deploys a contract and calls `wbtc::new` with all metadata. If this deploy and initialization were done using [NEAR CLI](https://docs.near.org/docs/tools/near-cli) the command would be:
+### SVG vs. PNG
```
Comment (robert-zaremba) [inline specs/Standards/Tokens/FungibleTokenMetadata.md:40]: again, I don't think we should standarize it (themes, file format, dimensions ....), especially on-chain. I would prefer to have a _Recommendation_ section where we put all that considerations. This is not needed for consensus. What we want to recommend is best practice in terms of usability and censorship resistance.

**Review by robert-zaremba — COMMENTED** [review]
I think we are going too far with standarizing things on chain. Nevertheless good job with the PR.

File: specs/Standards/Tokens/FungibleTokenMetadata.md [hunk 4]
```diff
@@ -67,33 +52,125 @@ type FungibleTokenMetadata = {
...
+
+### Light theme vs. dark theme
+
+Two different icons needed: `defaultIcon` (black text on white background) for iOS, websites. `darkIcon` or websites supporting dark theme and Android UI. After discussing with Ethereum wallet vendors, there are multiple cases where icon looks good on one platform, but horrible on another, because of color bleed issues on the stroke lines of an icon. Token projects are not happy if their icon looks like amaeur.
+
+Thus, we should provide both light and dark icons by default.
+
+### SVG vs. PNG
+
+SVG is a Turing complete language with a [massive attack surface](https://www.fortinet.com/blog/threat-research/scalable-vector-graphics-attack-surface-anatomy). It may even contain embedded JavaScript. After discussing with infosec experts, I recommend SVG is not supported for icons because of decoding complexity and denial of service issues.
```
Comment (chadoh) [inline specs/Standards/Tokens/FungibleTokenMetadata.md:116]: This link greatly overstates the JS-in-SVG attack vector https://security.stackexchange.com/q/212894

Need to finish reading it later today, but wanted to at least share that link for now.

**Comment by ori-near** [comment]
As the moderator, I assume this PR is no longer active given that there was no activity for over two months. Therefore, I am closing it. If anyone is interested in reopening this PR, please submit a new one.

**Comment by robert-zaremba** [comment]
Shall we reconsider cherry picking parts of this PR?

**Comment by frol** [comment]
@robert-zaremba If you believe the community will benefit from adding some of the mentioned improvements, absolutely, feel free to cherry-pick them and submit a new NEP extension 👍

**Pull request closed without merging**